Toolhouse.Monitoring/HttpRequestMetrics.cs

```diff
@@ -0,0 +1,148 @@
+using Prometheus;
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Toolhouse.Monitoring
+{
+    /// <summary>
+    /// Provides methods for instrumenting requests to an external API (e.g. via REST / SOAP).
+    /// </summary>
+    public sealed class HttpRequestMetrics
+    {
+        /// <summary>
+        /// Instruments a block of code that makes a request to an external API (e.g. via REST / SOAP).
+        /// </summary>
+        /// <param name="name">Descriptive name of thing being done. Should be a singular noun, e.g. "salesforce".</param>
+        /// <param name="request">An action which will perform the request</param>
+        public static void Instrument(string name, Action request)
+        {
+            InstrumentWithMetrics(name, requestMetrics =>
+            {
+                requestMetrics.PerformRequest(request);
+            });
+        }
+
+        /// <summary>
+        /// Instruments a block of code that makes a request to an external API (e.g. via REST / SOAP).
+        /// </summary>
+        /// <param name="name">Descriptive name of thing being done. Should be a singular noun, e.g. "salesforce".</param>
+        /// <param name="request">An action which will perform the request</param>
+        /// <remarks>A mutable Labels instance will be provided to the request.</remarks>
+        public static void Instrument(string name, Action<Labels> request)
```

Comment:
I feel like:

1. These `Instrument()` calls should be instance methods, not `static`
2. The constructor of this class should be `public` and accept a `name` argument

Reply:
I'd be interested to hear your thoughts on these points - especially the first one. My original idea to have these `static` was due to several things:

1. The original api being `static`
2. Ease of use of a `static` API
3. Not wanting to create an object that only has one data member (the name)

I think maybe a compromise might work well here - we could simply make the existing constructor `public`. That would allow for the pattern shown in the usage note above, but also allow for situations where one request type is executed in multiple places; where it'd be advantageous to share the `name`.

Reply:
Sorry, I'm coming back to this with fresh eyes and I think I've changed my mind a bit 😄.  My concern is mostly around the complexity of the API as a consumer. I think:

- Refactoring all of this into `HttpRequestMetrics` is 💯, but
- The `Metrics` class should be the only place that provides **static** methods for adding metrics to your app.

So I think what I'd like to see is these these various static `Instrument` overloads **moved** to the `Metrics` class (keep the old `InstrumentApiCall` marked as deprecated). That way, the main API for adding metrics is always available via the static `Metrics` class, but you can still `new` up an `HttpRequestMetrics` instance if you want to do something crazy.